pkg/kubectl/resource/builder.go

```diff
@@ -630,8 +614,23 @@ func (b *Builder) mappingFor(resourceOrKindArg string) (*meta.RESTMapping, error
 			return nil, err
 		}
 	}
+	if !gvk.Empty() {
+		return b.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	}
+
+	fullySpecifiedGVK, groupKind := schema.ParseKindArg(resourceOrKindArg)
+	if fullySpecifiedGVK == nil {
+		gvk := groupKind.WithVersion("")
+		fullySpecifiedGVK = &gvk
+	}
+
+	if !fullySpecifiedGVK.Empty() {
+		if mapping, err := b.mapper.RESTMapping(fullySpecifiedGVK.GroupKind(), fullySpecifiedGVK.Version); err == nil {
+			return mapping, nil
+		}
+	}
 
-	return b.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
+	return b.mapper.RESTMapping(groupKind, "")
```

Comment:
if this errors out, you want to return `return b.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)`, right?

Reply:
Thanks, updated this